octoprint_mrbeam/static/js/app/view-models/settings/maintenance.js

```diff
@@ -196,6 +223,42 @@ $(function () {
                 });
             });
             self.updateSettingsAbout();
+
+            const clickableContainers = document.querySelectorAll(
+                ".prefilter-clickable"
+            );
+            // Add click event listeners to each container
+            clickableContainers.forEach((container) => {
+                container.addEventListener("click", (event) => {
+                    // Find the radio input element within the container
+                    const radioInput = container.querySelector(
+                        'input[type="radio"]'
+                    );
+                    const clickedElement = event.target;
+
+                    // Check if the clicked element is the "Buy now" link
+                    if (clickedElement.id === "prefilter_shop_link") {
+                        // Prevent the click event from propagating further
+                        event.stopPropagation();
+                    } else {
+                        self.heavyDutyPrefilter(
+                            radioInput.getAttribute("value")
+                        );
+                    }
+                });
+            });
```

Comment:
This is quite unclear; What are we trying to do here?
And maybe we can move this chunk and the one below into different functions for readability purposes so that self.onAllBound is more clear

Reply:
the purpose is to make the whole div clickable to change the radio selection instead just the radio input it selfe

Reply:
Yea maybe just move this chunk and the one below into separate methods with readable names for more clarity